cmake/modules/user_cache.cmake

```diff
@@ -0,0 +1,99 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2021, Nordic Semiconductor ASA
+
+# Configure user cache directory.
+#
+# The user cache can be used for caching of data that should be persistent
+# across builds to speed up CMake configure / build system generation and/or
+# compilation.
+#
+# Only data that can be safely re-generated should be placed in this cache.
+#
+# Zephyr build system uses this user cache to store Zephyr compiler check
+# results which significantly improve toolchain testing performance.
+# See https://github.com/zephyrproject-rtos/zephyr/pull/7102 for details.
+#
+# Outcome:
+# The following variables will be defined when this CMake module completes:
+#
+# - USER_CACHE_DIR: User cache directory in use.
+#
+# If the above variable is already set when this CMake module is loaded,
+# then no changes to the variable will happen.
+#
+# CMake module dependencies:
+# - python CMake module.
```

Comment:
Isn't this documented well enough from the 

include(python)

statement?

We don't say

// This C file uses bluetooth

#include <bluetooth.h>

Reply:
This is added because i'm currently doing some additional work on preparing rst doc generation so this information can be automatically included in our doc-set.

That way users can get an overview of Zephyr CMake modules and their description, including dependencies which then can provide a link to the relevant Zephyr CMake module description.

Reply:
Could you keep the rst patches out of this PR.

Now we are inconsistently documenting module dependencies, which is worse than no docs.

Reply:
> Could you keep the rst patches out of this PR.

There are no rst patches in this PR.
I'm simply telling you why it make sense to have this described.

But fair enough if you prefer that we don't describe this at all.

Reply:
I prefer that we don't document obvious things like this.

Reply:
all occurrences have been removed